### PR TITLE
Ability to switch between back and front-facing camera

### PR DESCRIPTION
**Pull request by izzyaf**
As title, this PR allows this library to switch between front and back camera.
Also fixes the issue where the flashlight button shows up as "F...h" instead of the icon.

This will resolve these issues:
- https://github.com/AmolGangadhare/flutter_barcode_scanner/issues/46
- https://github.com/AmolGangadhare/flutter_barcode_scanner/issues/97

### CODE REVIEW

File: android/.idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="CompilerConfiguration">
+    <bytecodeTargetLevel target="1.8" />
+  </component>
+</project>
```
Comment (izzyaf) [inline android/.idea/compiler.xml:6]: These are artifacts made by opening example app using latest Android Studio.

File: android/.idea/gradle.xml [hunk 2]
```diff
@@ -14,6 +14,7 @@
           </set>
         </option>
         <option name="resolveModulePerSourceSet" value="false" />
+        <option name="useQualifiedModuleNames" value="true" />
```
Comment (izzyaf) [inline android/.idea/gradle.xml:17]: These are artifacts made by opening example app using latest Android Studio.

File: android/.idea/jarRepositories.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      <option name="name" value="BintrayJCenter" />
+      <option name="url" value="https://jcenter.bintray.com/" />
+    </remote-repository>
+    <remote-repository>
+      <option name="id" value="Google" />
+      <option name="name" value="Google" />
+      <option name="url" value="https://dl.google.com/dl/android/maven2/" />
+    </remote-repository>
+  </component>
+</project>
```
Comment (izzyaf) [inline android/.idea/jarRepositories.xml:25]: These are artifacts made by opening example app using latest Android Studio.

File: android/.idea/misc.xml [hunk 4]
```diff
@@ -5,7 +5,7 @@
       <configuration PROFILE_NAME="Debug" CONFIG_NAME="Debug" />
     </configurations>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (izzyaf) [inline android/.idea/misc.xml:8]: These are artifacts made by opening example app using latest Android Studio.

File: android/.idea/modules.xml [hunk 5]
```diff
@@ -2,7 +2,7 @@
 <project version="4">
   <component name="ProjectModuleManager">
     <modules>
-      <module fileurl="file://$PROJECT_DIR$/flutter_barcode_scanner.iml" filepath="$PROJECT_DIR$/flutter_barcode_scanner.iml" group="flutter_barcode_scanner" />
+      <module fileurl="file://$PROJECT_DIR$/.idea/modules/flutter_barcode_scanner.iml" filepath="$PROJECT_DIR$/.idea/modules/flutter_barcode_scanner.iml" />
```
Comment (izzyaf) [inline android/.idea/modules.xml:5]: These are artifacts made by opening example app using latest Android Studio.

File: android/src/main/res/drawable/ic_switch_camera.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24"
+    android:tint="?attr/colorControlNormal">
+  <path
+      android:fillColor="#ffffff"
+      android:pathData="M20,5h-3.17L15,3H9L7.17,5H4C2.9,5 2,5.9 2,7v12c0,1.1 0.9,2 2,2h16c1.1,0 2,-0.9 2,-2V7C22,5.9 21.1,5 20,5zM12,18c-2.76,0 -5,-2.24 -5,-5H5l2.5,-2.5L10,13H8c0,2.21 1.79,4 4,4c0.58,0 1.13,-0.13 1.62,-0.35l0.74,0.74C13.65,17.76 12.86,18 12,18zM16.5,15.5L14,13h2c0,-2.21 -1.79,-4 -4,-4c-0.58,0 -1.13,0.13 -1.62,0.35L9.64,8.62C10.35,8.24 11.14,8 12,8c2.76,0 5,2.24 5,5h2L16.5,15.5z"/>
+</vector>
```
Comment (izzyaf) [inline android/src/main/res/drawable/ic_switch_camera.xml:10]: Icon source: https://fonts.google.com/icons?selected=Material%20Icons%3Aflip_camera_ios

File: ios/Classes/SwiftFlutterBarcodeScannerPlugin.swift [hunk 7]
```diff
@@ -195,12 +195,23 @@ class BarcodeScannerViewController: UIViewController {
         flashButton.setTitle("Flash",for:.normal)
         flashButton.translatesAutoresizingMaskIntoConstraints=false
         
-        flashButton.setImage(UIImage(named: "ic_flash_off", in: Bundle(identifier: "org.cocoapods.flutter-barcode-scanner"), compatibleWith: nil),for:.normal)
+        flashButton.setImage(UIImage(named: "ic_flash_off", in: Bundle(for: SwiftFlutterBarcodeScannerPlugin.self), compatibleWith: nil),for:.normal)
```
Comment (izzyaf) [inline ios/Classes/SwiftFlutterBarcodeScannerPlugin.swift:198]: I find `SwiftFlutterBarcodeScannerPlugin.self` being a better identifier than relying on the previous string.

File: ios/flutter_barcode_scanner.podspec [hunk 8]
```diff
@@ -14,6 +14,7 @@ A new Flutter plugin supports barcode scanning on both Android and iOS.
   s.source           = { :path => '.' }
   s.source_files = 'Classes/**/*.{swift,h,m}'
   s.public_header_files = 'Classes/**/*.h'
+  s.resources = 'Assets/*.png'
```
Comment (izzyaf) [inline ios/flutter_barcode_scanner.podspec:17]: This line is important. Those image assets are not bundled with the plugin without this one.

**Comment by izzyaf** [comment]
@AmolGangadhare Can I get a review please?

**Review by MichaelDark — APPROVED** [review]
Finally, PR i was looking for, thank you 👍🏻

File: ios/Classes/SwiftFlutterBarcodeScannerPlugin.swift [hunk 9]
```diff
@@ -361,15 +374,21 @@ class BarcodeScannerViewController: UIViewController {
...
+        
+        switchCameraButton.translatesAutoresizingMaskIntoConstraints = false
+        // A little bit to the right.
+        switchCameraButton.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 10).isActive = true
+        switchCameraButton.heightAnchor.constraint(equalToConstant: 70.0).isActive = true
+        switchCameraButton.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: 0).isActive = true
     }
     
     /// Flash button click event listener
     @IBAction private func flashButtonClicked() {
```
Comment (AmolGangadhare) [inline ios/Classes/SwiftFlutterBarcodeScannerPlugin.swift:386]: Please handle the flash camera click when the front camera is open and the user clicks the flash icon. Currently the front camera freezes when user clicks flash icon

File: android/src/main/res/layout/barcode_capture.xml [hunk 10]
```diff
@@ -26,6 +26,21 @@
         android:layout_alignParentBottom="true"
         android:background="@android:color/black">
```
Comment (AmolGangadhare) [inline android/src/main/res/layout/barcode_capture.xml:28]: The layout component needs some fix, the flash icon click also reverses camera
I would suggest
```
  <ImageView
            android:id="@+id/imgViewSwitchCamera"
            android:layout_width="40dp"
            android:layout_height="40dp"
            android:layout_alignParentLeft="true"
            android:layout_alignParentBottom="true"
            android:layout_marginStart="10dp"
            android:layout_marginLeft="10dp"
            android:contentDescription="Reverse camera"
            android:src="@drawable/ic_switch_camera" />
```

**Review by AmolGangadhare — CHANGES_REQUESTED** [review]
I have reviewed the changes, requesting you to please revisit the comments I've added

File: ios/Classes/SwiftFlutterBarcodeScannerPlugin.swift (hunk 9 above)
Comment (izzyaf) [reply ios/Classes/SwiftFlutterBarcodeScannerPlugin.swift:386]: Change pushed.

File: android/src/main/res/layout/barcode_capture.xml (hunk 10 above)
Comment (izzyaf) [reply android/src/main/res/layout/barcode_capture.xml:28]: What's the difference between using `ImageView` and `Button`?

btw changes pushed. Also I don't get the issue that camera gets reversed on flash icon click.

**Comment by izzyaf** [comment]
@AmolGangadhare I pushed the changes you requested. Please take a look again.

**Review by AmolGangadhare — APPROVED** [review]
LGTM

Thanks for the wonderful PR!

**Comment by moseskarunia** [comment]
Not a typical github PR comment I know, but thanks so much for this!

**Pull request merged**